pyroomacoustics/room.py

```diff
@@ -1164,4 +1204,55 @@ def extrude(self, height):
 
         Room.extrude(self, np.array([0., 0., height]))
 
+    @classmethod
+    def inv_sabine(self, t60, room_dim):
+        '''
+        Given desired T60, (shoebox) room dimension and sound speed,
+        computes the absorption coefficient (amplitude) and image source
+        order needed. The speed of sound used is the package wide default
+        (in :py:data:`pyroomacoustics.parameters.constants`).
+
+        Parameters
+        ----------
+        t60: float
+            Desired T60 (time it takes to go from full amplitude to 60 dB decay) in seconds
+        room_dim: list of floats
+            List of length 2 or 3 of the room side lengths
+
+        Returns
+        -------
+        absorption: float
+            The absorption coefficient (in amplitude domain, to be passed to Room constructor)
+        max_order: int
+            The maximum image source order necessary to achieve the desired T60
+        '''
+        c = constants.get('c')
+
+        # Finding image sources up to a maximum order creates a (possibly 3D) diamond
+        # like pile of (reflected) rooms. Now we need to find the image source model order
+        # so that reflections at a distance of at least up to ``c * rt60`` are included.
+        # One possibility is to find the largest sphere (or circle in 2D) that fits in the
+        # diamond. This is what we are doing here.
+        R = []
+        for l1, l2 in  itertools.combinations(room_dim, 2):
+            R.append(l1 * l2 / np.sqrt(l1 ** 2 + l2 ** 2))
+
+        V = np.prod(room_dim)  # volume
+        # surface computation is diff for 2D and 3D
+        if len(room_dim) == 2:
+            S = np.sum(room_dim)
```

Comment:
Here I think it should be `2*np.sum(room_dim)` to sum the 2D room boundaries. This adds up just two of the walls.

Reply:
Right!